jctools-experimental/src/main/java/org/jctools/channels/proxy/DemoProxyResult.java

```diff
@@ -81,6 +83,8 @@ public void call2(float x, double y, boolean z) {
         long wOffset = this.writeAcquire();
         UnsafeAccess.UNSAFE.putFloat(wOffset + 4, x);
         UnsafeAccess.UNSAFE.putDouble(wOffset + 8, y);
+        // TODO: Should this offset not be +8 as the last value was a double?
+        // TODO: What about object references, does that depend on compressed oops being enabled?
```

Comment:
references require a parallel object references buffer.
You cannot (though some try) write refs into a byte array and reload them safely (because GC may move the ref).
You cannot (this one fails very quickly so I doubt people stick with it) write primitives into ref arrays because the JVM will try to treat these garbage pointers as pointers and crash.